eta/core/datasets.py

```diff
@@ -1219,14 +1178,36 @@ class LabeledDatasetBuilder(object):
     sampling, filtering by schema, and balance.
     '''
 
-    def __init__(self, dataset=LazyLabeledVideoDataset()):
-        self.dataset = dataset
+    def __init__(self, schema=None):
         self.transformers = []
+        self.entries = []
+        self.schema = schema
+
+    def add_entry(self, entry):
+        self.entries.append(entry)
+
+    def set_entries(self, entries):
+        self.entries = entries
+
+    def get_entries(self):
+        return self.entries
 
     def add_transform(self, transform):
         self.transformers.append(transform)
 
     def build(self):
         for transformer in self.transformers:
-            self.dataset = transformer.transform(self.dataset)
-        return self.dataset.make_dataset()
+            transformer.transform(self)
```

Comment:
Oh transformers take the builder in directly... hmm 
This is why putting too much responsibility in one object can be problematic, this could allow a transformer to add itself to the builder.  I think there should be a POJO (or POPO) that simply has the collection or records.  But make the interface for its construction and lifecycle internal to that of the builder:
 - Make builder
 - Make records
 - Add records to builder -> builder adds record to internal object
 - Builder passes internal collection to transformers, not self

Reply:
Right. I've made some changes.